### PR TITLE
Implement custom priorities.

### PR DESCRIPTION
**Pull request by mshima**
Fixes https://github.com/jhipster/generator-jhipster/issues/12338.

---

Please make sure the below checklist is followed for Pull Requests.

-   [ ] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
-   [ ] Tests are added where necessary
-   [ ] Documentation is added/updated where necessary
-   [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/master/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (bellow reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

### CODE REVIEW

**Comment by mshima** [comment]
@jhipster/developers suggestions to this one?
I am planning to merge.

**Comment by jdubois** [comment]
Wow, I've been struggling with those and you seem to be doing this so easily!!
Yes, we should have those better separated and documented, as it's probably very complicated for newcomers to understand.

**Comment by pascalgrimaud** [comment]
@mshima : good job as always. But it will be hard to review as your knowledges about this are far better than all of us. So I'm trusting you here.

Feel free to merge when you want.

**Comment by mshima** [comment]
@pascalgrimaud I was asking if there is suggestions about naming and order.
This will have a big impact on our development, so I think I is right to ask.
Anyway this can be changed before 7.0 release, merging.

**Comment by pascalgrimaud** [comment]
@mshima : will it impact existing blueprint ?

**Comment by mshima** [comment]
@pascalgrimaud at current state probably every blueprint is broken, due to past workflow changes.
For this change I created backward compatibility:
https://github.com/jhipster/generator-jhipster/blob/9036bbf61c9959a9b8c30c92332d71c55b6d0b42/generators/app/index.js#L403
https://github.com/jhipster/generator-jhipster/blob/9036bbf61c9959a9b8c30c92332d71c55b6d0b42/generators/app/index.js#L474

Can you release beta0 later this week?

**Comment by pascalgrimaud** [comment]
about release a beta version, it needs to be discussed with all the team before. If all agree, no problem

I'm a little bit late, as I have to do a patch release for v6 too... but too buzy at work and too much stuff to prepare for JHipster Code

**Comment by atomfrede** [comment]
Maybe we can do 6.x patch release and the first 7 beta on jhipster code?

File: generators/common/index.js [hunk 1]
```diff
@@ -72,18 +72,42 @@ module.exports = class extends BaseBlueprintGenerator {
...
                 this.loadAppConfig();
                 this.loadClientConfig();
                 this.loadServerConfig();
                 this.loadTranslationConfig();
             },
-            setupSharedOptions() {
+        };
+    }
+
+    get loading() {
```
Comment (murdos) [inline generators/common/index.js:86]: I think `loadingConfig` would be more explicit

Comment (mshima) [reply generators/common/index.js:86]: I am not sure that would make much difference.
Probably we should add a good description at `generator-blueprint.js` and the blueprint generator.
But if I am open for the change.

**Pull request merged**